Etabs_Adapter/CRUD/Read/Results/MeshResults.cs

```diff
@@ -508,14 +698,48 @@ private List<MeshStress> SmoothenStresses(List<MeshStress> forces)
                 double pr2 = group.Average(x => x.Principal_2);
                 double pr1_2 = group.Average(x => x.Principal_1_2);
 
-                smoothenedForces.Add(new MeshStress(first.ObjectId, first.NodeId, "", first.ResultCase, first.ModeNumber, first.TimeStep, first.MeshResultLayer, first.LayerPosition, MeshResultSmoothingType.ByPanel, first.Orientation, sxx, syy, sxy, txx, tyy, pr1, pr2, pr1_2));
+                smoothenedForces.Add(new MeshStress(first.ObjectId, first.NodeId, "", first.ResultCase, first.ModeNumber, first.TimeStep, 
+                                                        first.MeshResultLayer, first.LayerPosition, MeshResultSmoothingType.ByPanel, first.Orientation, 
+                                                        sxx, syy, sxy, txx, tyy, pr1, pr2, pr1_2));
             }
 
             return smoothenedForces;
         }
 
         /***************************************************/
 
+
+        private List<MeshVonMises> SmoothenVonMisesStresses(List<MeshVonMises> forces)
+        {
+            List<MeshVonMises> smoothenedVMStresses = new List<MeshVonMises>();
+
+            foreach (IEnumerable<MeshVonMises> group in forces.GroupBy(x => new { x.ResultCase, x.TimeStep, x.NodeId }))
+            {
+                MeshVonMises first = group.First();
+
+                double s = group.Average(x => x.S);
+                double n = group.Average(x => x.N);
+                double m = group.Average(x => x.M);
+
+                smoothenedVMStresses.Add(new MeshVonMises(first.ObjectId, first.NodeId, "", first.ResultCase, first.ModeNumber, first.TimeStep, 
+                                                            first.MeshResultLayer, first.LayerPosition, MeshResultSmoothingType.ByPanel, first.Orientation,s,n,m));
+            }
+
+            return smoothenedVMStresses;
+        }
+
+        /***************************************************/
+
+        private double ComputeVonMisesMoment(double svmTop, double svmBot, double thk)
+        {
+            double svmAvg = (svmTop + svmBot) / 2;
+            double vonMisesMoment = ((svmBot - svmAvg) * (thk / 2) * (1 / 2) * (thk - 2 * thk / 2 * 1 / 3)) / 1000;
+
+            return vonMisesMoment;
+        }
```

Comment:
```suggestion
```
As above

Reply:
@peterjamesnugent, code updated in line with comment